airflow/bin/cli.py

```diff
@@ -336,6 +336,47 @@ def list_tasks(args, dag=None):
         print("\n".join(sorted(tasks)))
 
 
+def parsedate_enhanced(args, dag):
+    '''
+    In addition to supporting date parsing, this also supports
+    @now,
+    @start_date (i.e. dag start date),
+    @end_date (i.e. dag end date if one exists), and
+    @latest (i.e. run closest to and before now that honors the schedule interval)
+    '''
+    timestr = args.execution_date
+    now = datetime.now()
+    start_date = dag.default_args['start_date'] if \
+        (dag.default_args and 'start_date' in dag.default_args) else None
+    end_date = dag.default_args['end_date'] if \
+        (dag.default_args and 'end_date' in dag.default_args) else None
+
+    if timestr == '@now':
+        return now
+    elif timestr == '@start_date':
+        if start_date:
+            return start_date
+        else:
+            raise Exception("{} does not have a start_date, so @start_date "
+                            "is not supported for this dag".format(dag.dag_id))
+    elif timestr == '@end_date':
+        if end_date:
+            return end_date
+        else:
+            raise Exception("{} does not have an end_date, so @end_date "
+                            "is not supported for this dag".format(dag.dag_id))
+    elif timestr == '@latest':
+        effective_start_date = start_date if start_date else now
+        effective_end_date = end_date if end_date else now
+        if dag.schedule_interval == '@once' or dag.schedule_interval == 'None':
+            return now
+        else:
+            return dag.date_range(start_date=effective_start_date,
+                                  end_date=effective_end_date)[-2]
```

Comment:
It's possible that this list has less than 2 elements, in which case this will raise an error